guava/src/com/google/common/math/BigIntegerMath.java

```diff
@@ -332,7 +337,10 @@ public static BigInteger divide(BigInteger p, BigInteger q, RoundingMode mode) {
    *
    * @throws IllegalArgumentException if {@code n < 0}
    */
-  public static BigInteger factorial(int n) {
+  @SuppressWarnings(value = {"lowerbound:assignment.type.incompatible",// line 350: Since `n` is required to be non negative
```

Comment:
This is minor, but try to avoid lines that are longer than about 140 characters, because GitHub's PR display cuts them off.

Reply:
Also line numbers here.